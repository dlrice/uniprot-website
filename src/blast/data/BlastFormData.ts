export type BlastFormValue = {
  fieldName: string;
  selected?: string | undefined;
  selectedLabel?: string | undefined;
  type: BlastFieldTypes;
  values?: { label?: string; value?: string | number }[];
};

export enum BlastFields {
  program = 'Program',
  stype = 'Sequence type',
  email = 'Email',
  sequence = 'Sequence',
  taxon = 'Taxon',
  targetDb = 'Target database',
  threshold = 'E-Threshold',
  matrix = 'Matrix',
  filter = 'Filter',
  gapped = 'Gapped',
  hits = 'Hits',
}

export enum BlastFieldTypes {
  textarea,
  select,
  autocomplete,
}

export type BlastFormValues = { [x in BlastFields]: BlastFormValue };

export default {
  [BlastFields.program]: {
    fieldName: 'program',
    selected: 'blastp',
  },
  [BlastFields.stype]: {
    fieldName: 'stype',
    selected: 'protein',
  },
  [BlastFields.email]: {
    fieldName: 'email',
    selected: 'uuw_dev@uniprot.org',
  },
  [BlastFields.sequence]: {
    fieldName: 'sequence',
    type: BlastFieldTypes.textarea,
  },
  [BlastFields.targetDb]: {
    fieldName: 'database',
    type: BlastFieldTypes.select,
    selected: 'uniprotkb_refprotswissprot',
    values: [
      {
        value: 'uniprotkb_refprotswissprot',
        label: 'UniProtKb',
      },
    ],
  },
  [BlastFields.taxon]: {
    fieldName: 'taxids',
    type: BlastFieldTypes.autocomplete,
  },
  [BlastFields.threshold]: {
    fieldName: 'threshold',
    type: BlastFieldTypes.select,
    selected: '10',
    values: [
      { value: '0.0001' },
      { value: '0.001' },
      { value: '0.01' },
      { value: '0.1' },
      { value: '1' },
      { value: '10' },
      { value: '100' },
      { value: '1000' },
    ],
  },
  [BlastFields.matrix]: {
    fieldName: 'matrix',
    type: BlastFieldTypes.select,
    selected: null,
    values: [
      {
        value: '',
        label: 'Auto',
      },
      { value: '45' },
      { value: '62' },
      { value: '80' },
      { value: '70' },
      { value: '30' },
    ],
  },
  [BlastFields.filter]: {
    fieldName: 'filter',
    type: BlastFieldTypes.select,
    selected: 'F',
    values: [
      { value: 'F', label: 'None' },
      { valueNone: 'T', label: 'Filter low complexity regions' },
      { value: 'mask', label: 'Mask lookup table only' },
    ],
  },
  [BlastFields.gapped]: {
    fieldName: 'gapped',
    type: BlastFieldTypes.select,
    selected: 'true',
    values: [
      { value: 'true', label: 'yes' },
      { value: 'false', label: 'no' },
    ],
  },
  [BlastFields.hits]: {
    fieldName: 'hits',
    type: BlastFieldTypes.select,
    selected: '250',
    values: [
      { value: '50' },
      { value: '100' },
      { value: '250' },
      { value: '500' },
      { value: '750' },
      { value: '1000' },
    ],
  },
};
