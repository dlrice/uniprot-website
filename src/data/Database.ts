import { Database, DatabaseCategory } from '../model/types/databaseTypes';
import EntrySection from '../model/types/EntrySection';

export const databaseCategoryToString = new Map<DatabaseCategory, string>([
  [DatabaseCategory.SEQUENCE, 'Sequence databases'],
  [DatabaseCategory.STRUCTURE, '3D structure databases'],
  [DatabaseCategory.INTERACTION, 'Protein-protein interaction databases'],
  [DatabaseCategory.CHEMISTRY, 'Chemistry'],
  [DatabaseCategory.FAMILY, 'Protein family/group databases'],
  [DatabaseCategory.PTM, 'PTM databases'],
  [DatabaseCategory.POLYMORPHISM, 'Polymorphism and mutation databases'],
  [DatabaseCategory.GEL, '2D gel databases'],
  [DatabaseCategory.PROTEOMIC, 'Proteomic databases'],
  [DatabaseCategory.PROTOCOL, 'Protocols and materials databases'],
  [DatabaseCategory.GENOME, 'Genome annotation databases'],
  [DatabaseCategory.ORGANISM, 'Organism-specific databases'],
  [DatabaseCategory.PHYLOGENOMIC, 'Phylogenomic databases'],
  [DatabaseCategory.PATHWAY, 'Enzyme and pathway databases'],
  [DatabaseCategory.EXPRESSION, 'Gene expression databases'],
  [DatabaseCategory.DOMAIN, 'Family and domain databases'],
  [DatabaseCategory.OTHER, 'Other'],
]);

const databaseCategoryToDatabases = new Map<DatabaseCategory, Database[]>();
databaseCategoryToDatabases.set(DatabaseCategory.SEQUENCE, [
  // Not found: EMBL/GenBank/DDBJ
  Database.EMBL,
  Database.CCDS,
  Database.PIR,
  Database.RefSeq,
  Database.UniGene,
]);
databaseCategoryToDatabases.set(DatabaseCategory.STRUCTURE, [
  Database.PDB,
  Database.SMR,
  Database.DisProt,
  Database.ProteinModelPortal,
]);
databaseCategoryToDatabases.set(DatabaseCategory.INTERACTION, [
  Database.BioGrid,
  Database.DIP,
  Database.IntAct,
  Database.MINT,
  Database.STRING,
]);
databaseCategoryToDatabases.set(DatabaseCategory.CHEMISTRY, [
  Database.BindingDB,
  Database.ChEMBL,
  Database.DrugBank,
  Database.GuidetoPHARMACOLOGY,
]);
databaseCategoryToDatabases.set(DatabaseCategory.FAMILY, [
  // Not found: Database.PptaseDB,
  Database.Allergome,
  Database.CAZy,
  Database.MEROPS,
  Database.MoonProt,
  Database.mycoCLAP,
  Database.PeroxiBase,
  Database.REBASE,
  Database.TCDB,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PTM, [
  // Not found: DEPOD, PhosphoSite, PhosSite
  Database.DEPOD,
  Database.PhosphoSitePlus,
  Database.UniCarbKB,
]);
databaseCategoryToDatabases.set(DatabaseCategory.POLYMORPHISM, [
  Database.dbSNP,
  Database.DMDM,
]);
databaseCategoryToDatabases.set(DatabaseCategory.GEL, [
  Database.SWISS_2DPAGE,
  Database.COMPLUYEAST_2DPAGE,
  Database.DOSAC_COBS_2DPAGE,
  Database.OGP,
  Database.REPRODUCTION_2DPAGE,
  Database.UCD_2DPAGE,
  Database.World_2DPAGE,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PROTEOMIC, [
  Database.MaxQB,
  Database.PaxDb,
  Database.PeptideAtlas,
  Database.PRIDE,
  Database.ProMEX,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PROTOCOL, [Database.DNASU]);
databaseCategoryToDatabases.set(DatabaseCategory.GENOME, [
  Database.Ensembl,
  Database.EnsemblBacteria,
  Database.EnsemblFungi,
  Database.EnsemblMetazoa,
  Database.EnsemblPlants,
  Database.EnsemblProtists,
  Database.GeneCards,
  Database.GeneID,
  Database.KEGG,
  Database.PATRIC,
  Database.UCSC,
  Database.VectorBase,
]);
databaseCategoryToDatabases.set(DatabaseCategory.ORGANISM, [
  Database.ArachnoServer,
  Database.CGD,
  Database.ConoServer,
  Database.CTD,
  Database.dictyBase,
  Database.EchoBASE,
  Database.EcoGene,
  Database.euHCVdb,
  Database.EuPathDB,
  Database.FlyBase,
  Database.GeneReviews,
  Database.Gramene,
  Database.H_InvDB,
  Database.HGNC,
  Database.HPA,
  Database.LegioList,
  Database.Leproma,
  Database.MaizeGDB,
  Database.MIM,
  Database.MGI,
  Database.neXtProt,
  Database.Orphanet,
  Database.PharmGKB,
  Database.PomBase,
  Database.PseudoCAP,
  Database.RGD,
  Database.SGD,
  Database.TAIR,
  Database.TubercuList,
  Database.WormBase,
  Database.Xenbase,
  Database.ZFIN,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PHYLOGENOMIC, [
  Database.eggNOG,
  Database.GeneTree,
  Database.HOGENOM,
  Database.HOVERGEN,
  Database.InParanoid,
  Database.KO,
  Database.OMA,
  Database.OrthoDB,
  Database.PhylomeDB,
  Database.TreeFam,
]);
databaseCategoryToDatabases.set(DatabaseCategory.PATHWAY, [
  Database.BioCyc,
  Database.BRENDA,
  Database.Reactome,
  Database.SABIO_RK,
  Database.SignaLink,
  Database.UniPathway,
]);
databaseCategoryToDatabases.set(DatabaseCategory.EXPRESSION, [
  Database.Bgee,
  Database.ExpressionAtlas,
]);
databaseCategoryToDatabases.set(DatabaseCategory.DOMAIN, [
  Database.HAMAP,
  Database.InterPro,
  Database.Gene3D,
  Database.PANTHER,
  Database.Pfam,
  Database.PIRSF,
  Database.PRINTS,
  Database.ProDom,
  Database.SMART,
  Database.SUPFAM,
  Database.TIGRFAMs,
  Database.PROSITE,
]);
databaseCategoryToDatabases.set(DatabaseCategory.OTHER, [
  // Not found: NextBio
  Database.ChiTaRS,
  Database.EvolutionaryTrace,
  Database.GenomeRNAi,
  Database.GeneWiki,
  Database.PMAP_CutDB,
  Database.PRO,
]);

const entrySectionToDatabaseCategories = new Map<
  EntrySection,
  DatabaseCategory[]
>();
entrySectionToDatabaseCategories.set(EntrySection.Expression, [
  DatabaseCategory.ORGANISM,
  DatabaseCategory.EXPRESSION,
]);
entrySectionToDatabaseCategories.set(EntrySection.FamilyAndDomains, [
  DatabaseCategory.FAMILY,
  DatabaseCategory.PHYLOGENOMIC,
  DatabaseCategory.DOMAIN,
]);
entrySectionToDatabaseCategories.set(EntrySection.Function, [
  DatabaseCategory.PATHWAY,
  DatabaseCategory.FAMILY,
]);
entrySectionToDatabaseCategories.set(EntrySection.Interaction, [
  DatabaseCategory.INTERACTION,
  DatabaseCategory.CHEMISTRY,
]);
entrySectionToDatabaseCategories.set(EntrySection.NamesAndTaxonomy, [
  DatabaseCategory.ORGANISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.PathologyOrBioTech, [
  DatabaseCategory.ORGANISM,
  DatabaseCategory.OTHER,
  DatabaseCategory.FAMILY,
  DatabaseCategory.CHEMISTRY,
  DatabaseCategory.POLYMORPHISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.ProteinProcessing, [
  DatabaseCategory.PROTEOMIC,
  DatabaseCategory.GEL,
  DatabaseCategory.PTM,
  DatabaseCategory.OTHER,
]);
entrySectionToDatabaseCategories.set(EntrySection.Sequence, [
  DatabaseCategory.SEQUENCE,
  DatabaseCategory.GENOME,
  DatabaseCategory.POLYMORPHISM,
]);
entrySectionToDatabaseCategories.set(EntrySection.Structure, [
  DatabaseCategory.STRUCTURE,
  DatabaseCategory.PROTOCOL,
  DatabaseCategory.OTHER,
]);

// Combine multiple maps into single map for quick read access.
export const entrySectionDatabaseToDatabaseCategory = new Map<
  string,
  DatabaseCategory
>();
for (const [
  entrySection,
  databaseCategories,
] of entrySectionToDatabaseCategories) {
  if (databaseCategories) {
    for (const databaseCategory of databaseCategories) {
      const databases = databaseCategoryToDatabases.get(databaseCategory);
      if (databases) {
        for (const database of databases) {
          entrySectionDatabaseToDatabaseCategory.set(
            `${entrySection}_${database}`,
            databaseCategory
          );
        }
      }
    }
  }
}

export default entrySectionDatabaseToDatabaseCategory;
