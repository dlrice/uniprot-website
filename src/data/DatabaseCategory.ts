enum databaseCategory {
  SEQUENCE = 'SEQUENCE',
  STRUCTURE = 'STRUCTURE',
  INTERACTION = 'INTERACTION',
  CHEMISTRY = 'CHEMISTRY',
  FAMILY = 'FAMILY',
  PTM = 'PTM',
  POLYMORPHISM = 'POLYMORPHISM',
  GEL = 'GEL',
  PROTEOMIC = 'PROTEOMIC',
  PROTOCOL = 'PROTOCOL',
  GENOME = 'GENOME',
  ORGANISM = 'ORGANISM',
  PHYLOGENOMIC = 'PHYLOGENOMIC',
  PATHWAY = 'PATHWAY',
  EXPRESSION = 'EXPRESSION',
  DOMAIN = 'DOMAIN',
  OTHER = 'OTHER',
}

export default databaseCategory;
