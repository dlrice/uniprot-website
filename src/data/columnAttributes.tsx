const small = 200; //10rem;
const medium = 400; //15rem;
const large = 600; //20rem;

export type columnAttributes = {
  [columnName: string]: {
    width: number;
    label: string;
  };
};

const columnAttributes: columnAttributes = {
  accession: {
    width: small,
    label: 'Entry',
  },
  id: {
    width: small,
    label: 'Entry Name',
  },
  protein_name: {
    width: large,
    label: 'Protein names',
  },
  gene_names: {
    width: medium,
    label: 'Gene Names',
  },
  organism: {
    width: large,
    label: 'Organism',
  },
};

export default columnAttributes;
