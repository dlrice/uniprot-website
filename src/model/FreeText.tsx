import React, { Fragment } from 'react';
import UniProtEvidenceTag from '../components/UniProtEvidenceTag';
import { EvidenceType } from './types/modelTypes';

export enum FreeTextType {
  FUNCTION = 'FUNCTION',
  CATALYTIC_ACTIVITY = 'CATALYTIC ACTIVITY',
  COFACTOR = 'COFACTOR',
  ACTIVITY_REGULATION = 'ACTIVITY REGULATION',
  BIOPHYSICOCHEMICAL_PROPERTIES = 'BIOPHYSICOCHEMICAL PROPERTIES',
  PATHWAY = 'PATHWAY',
  SUBUNIT = 'SUBUNIT',
  INTERACTION = 'INTERACTION',
  SUBCELLULAR_LOCATION = 'SUBCELLULAR LOCATION',
  ALTERNATIVE_PRODUCTS = 'ALTERNATIVE PRODUCTS',
  TISSUE_SPECIFICITY = 'TISSUE SPECIFICITY',
  DEVELOPMENTAL_STAGE = 'DEVELOPMENTAL STAGE',
  INDUCTION = 'INDUCTION',
  DOMAIN = 'DOMAIN',
  PTM = 'PTM',
  RNA_EDITING = 'RNA EDITING',
  MASS_SPECTROMETRY = 'MASS SPECTROMETRY',
  POLYMORPHISM = 'POLYMORPHISM',
  DISEASE = 'DISEASE',
  DISRUPTION_PHENOTYPE = 'DISRUPTION PHENOTYPE',
  ALLERGEN = 'ALLERGEN',
  TOXIC_DOSE = 'TOXIC DOSE',
  BIOTECHNOLOGY = 'BIOTECHNOLOGY',
  PHARMACEUTICAL = 'PHARMACEUTICAL',
  MISCELLANEOUS = 'MISCELLANEOUS',
  SIMILARITY = 'SIMILARITY',
  CAUTION = 'CAUTION',
  SEQUENCE_CAUTION = 'SEQUENCE CAUTION',
  WEBRESOURCE = 'WEBRESOURCE',
  UNKNOWN = 'UNKNOWN',
}

export type FreeTextData = {
  comments?: [
    {
      commentType: FreeTextType;
      texts: [{ value: string; evidences: EvidenceType[] }];
    }
  ];
};

type FreeTextDataProps = {
  data: FreeTextData;
  type: FreeTextType;
  includeTitle?: boolean;
};

export const FreeText: React.FC<FreeTextDataProps> = ({
  data,
  type,
  includeTitle = false,
}) => {
  if (!data.comments || !data.comments.some(d => d.commentType === type)) {
    return null;
  }
  const freeTextData = data.comments
    .filter(d => d.commentType === type)
    .map((item, i) => (
      <p key={`freetext_${i}_${type}`}>
        {item.texts.map((itemText, j) => {
          return (
            <Fragment key={`freetext_${i}_${type}_${j}`}>
              {itemText.value}
              {itemText.evidences.map(evidence => (
                <UniProtEvidenceTag evidence={evidence} key={evidence.id} />
              ))}
            </Fragment>
          );
        })}
      </p>
    ));

  return (
    <Fragment>
      {includeTitle && (
        <h4 style={{ textTransform: 'capitalize' }}>{type.toLowerCase()}</h4>
      )}
      {freeTextData}
    </Fragment>
  );
};
