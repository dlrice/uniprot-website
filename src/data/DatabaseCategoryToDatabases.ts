import DatabaseCategory from './DatabaseCategory';

const databaseCategoryToDatabases = new Map<DatabaseCategory, string[]>([
  [
    DatabaseCategory.SEQUENCE,
    ['EMBL/GenBank/DDBJ', 'CCDS', 'PIR', 'RefSeq', 'UniGene'],
  ],
  [DatabaseCategory.STRUCTURE, ['PDB', 'SMR', 'DisProt', 'ProteinModelPortal']],
  [
    DatabaseCategory.INTERACTION,
    ['BioGrid', 'DIP', 'IntAct', 'MINT', 'STRING'],
  ],
  [
    DatabaseCategory.CHEMISTRY,
    ['BindingDB', 'ChEMBL', 'DrugBank', 'GuidetoPHARMACOLOGY'],
  ],
  [
    DatabaseCategory.FAMILY,
    [
      'Allergome',
      'CAZy',
      'MEROPS',
      'MoonProt',
      'mycoCLAP',
      'PeroxiBase',
      'PptaseDB',
      'REBASE',
      'TCDB',
    ],
  ],
  [DatabaseCategory.PTM, ['DEPOD', 'PhosphoSite', 'PhosSite', 'UniCarbKB']],
  [DatabaseCategory.POLYMORPHISM, ['dbSNP', 'DMDM']],
  [
    DatabaseCategory.GEL,
    [
      'SWISS-2DPAGE',
      'COMPLUYEAST-2DPAGE',
      'DOSAC-COBS-2DPAGE',
      'OGP',
      'REPRODUCTION-2DPAGE',
      'UCD-2DPAGE',
      'World-2DPAGE',
    ],
  ],
  [
    DatabaseCategory.PROTEOMIC,
    ['MaxQB', 'PaxDb', 'PeptideAtlas', 'PRIDE', 'ProMEX'],
  ],
  [DatabaseCategory.PROTOCOL, ['DNASU']],
  [
    DatabaseCategory.GENOME,
    [
      'Ensembl and Ensembl Genomes',
      'Ensembl',
      'EnsemblBacteria',
      'EnsemblFungi',
      'EnsemblMetazoa',
      'EnsemblPlants',
      'EnsemblProtists',
      'GeneCards',
      'GeneID',
      'KEGG',
      'PATRIC',
      'UCSC',
      'VectorBase',
    ],
  ],
  [
    DatabaseCategory.ORGANISM,
    [
      'ArachnoServer',
      'CGD',
      'ConoServer',
      'CTD',
      'dictyBase',
      'EchoBASE',
      'EcoGene',
      'euHCVdb',
      'EuPathDB',
      'FlyBase',
      'GeneReviews',
      'Gramene',
      'H-InvDB',
      'HGNC',
      'HPA',
      'LegioList',
      'Leproma',
      'MaizeGDB',
      'MIM',
      'MGI',
      'neXtProt',
      'Orphanet',
      'PharmGKB',
      'PomBase',
      'PseudoCAP',
      'RGD',
      'SGD',
      'TAIR',
      'TubercuList',
      'WormBase',
      'Xenbase',
      'ZFIN',
    ],
  ],
  [
    DatabaseCategory.PHYLOGENOMIC,
    [
      'eggNOG',
      'GeneTree',
      'HOGENOM',
      'HOVERGEN',
      'InParanoid',
      'KO',
      'OMA',
      'OrthoDB',
      'PhylomeDB',
      'TreeFam',
    ],
  ],
  [
    DatabaseCategory.PATHWAY,
    ['BioCyc', 'BRENDA', 'Reactome', 'SABIO-RK', 'SignaLink', 'UniPathway'],
  ],
  [DatabaseCategory.EXPRESSION, ['Bgee', 'ExpressionAtlas']],
  [
    DatabaseCategory.DOMAIN,
    [
      'HAMAP',
      'InterPro',
      'Gene3D',
      'PANTHER',
      'Pfam',
      'PIRSF',
      'PRINTS',
      'ProDom',
      'SMART',
      'SUPFAM',
      'TIGRFAMs',
      'PROSITE',
    ],
  ],
  [
    DatabaseCategory.OTHER,
    [
      'ChiTaRS',
      'EvolutionaryTrace',
      'GenomeRNAi',
      'GeneWiki',
      'NextBio',
      'PMAP-CutDB',
      'PRO',
    ],
  ],
]);

export default databaseCategoryToDatabases;
