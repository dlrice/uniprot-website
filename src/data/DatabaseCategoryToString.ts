import DatabaseCategory from './DatabaseCategory';

const databaseCategoryToDatabases = new Map<DatabaseCategory, string>([
  [DatabaseCategory.SEQUENCE, 'Sequence databases'],
  [DatabaseCategory.STRUCTURE, '3D structure databases'],
  [DatabaseCategory.INTERACTION, 'Protein-protein interaction databases'],
  [DatabaseCategory.CHEMISTRY, 'Chemistry'],
  [DatabaseCategory.FAMILY, 'Protein family/group databases'],
  [DatabaseCategory.PTM, 'PTM databases'],
  [DatabaseCategory.POLYMORPHISM, 'Polymorphism and mutation databases'],
  [DatabaseCategory.GEL, '2D gel databases'],
  [DatabaseCategory.PROTEOMIC, 'Proteomic databases'],
  [DatabaseCategory.PROTOCOL, 'Protocols and materials databases'],
  [DatabaseCategory.GENOME, 'Genome annotation databases'],
  [DatabaseCategory.ORGANISM, 'Organism-specific databases'],
  [DatabaseCategory.PHYLOGENOMIC, 'Phylogenomic databases'],
  [DatabaseCategory.PATHWAY, 'Enzyme and pathway databases'],
  [DatabaseCategory.EXPRESSION, 'Gene expression databases'],
  [DatabaseCategory.DOMAIN, 'Family and domain databases'],
  [DatabaseCategory.OTHER, 'Other'],
]);

export default databaseCategoryToDatabases;
