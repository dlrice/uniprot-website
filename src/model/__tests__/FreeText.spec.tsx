import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import { FreeText, FreeTextType } from '../FreeText';
import data from './modelData.json';

jest.mock('uuid/v1', () => () => 0);
configure({ adapter: new Adapter() });

describe('FreeText component', () => {
  test('should render free text CC', () => {
    const wrapper = shallow(
      <FreeText data={data} type={FreeTextType.DISRUPTION_PHENOTYPE} />
    );
    expect(wrapper).toMatchSnapshot();
  });
});
