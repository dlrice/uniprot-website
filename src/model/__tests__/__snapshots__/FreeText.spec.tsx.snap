// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FreeText component should render free text CC 1`] = `
"<Fragment>
  <p>
    value
    <UniProtEvidence evidence={{...}} />
  </p>
</Fragment>"
`;
